syntax = "proto3";

// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. chat_service.proto

package chat;

// Define Chat Service & Message Types

service ChatService {
  rpc SendMessage (SendRequest) returns (SendResponse) {}
  rpc ReceiveMessage (User) returns (stream ChatMessage) {}
  rpc GetUsers (Empty) returns (stream User) {}
}

message SendRequest {
  string sender = 1;
  string recipient = 2;
  string content = 3;
}

message SendResponse {
  bool success = 1;
  string message = 2;
}

message Empty {}

message ChatMessage {
  string sender = 1;
  string content = 2;
}

message User {
  string username = 1;
}

// Define Auth Service & Messages

service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse) {}
  rpc Register (RegisterRequest) returns (RegisterResponse) {}
  rpc Delete (DeleteRequest) returns (DeleteResponse) {}
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message DeleteRequest {
  string username = 1;
}

message DeleteResponse {
  bool success = 1;
  string message = 2;
}
